metal/end_model/end_model.py

```diff
@@ -72,7 +71,9 @@ def _build(self, input_module, middle_modules, head_module):
             self.network = nn.Sequential(input_layer, *middle_layers, head)
 
         # Construct loss module
-        self.criteria = SoftCrossEntropyLoss(reduction="sum")
+        self.criteria = SoftCrossEntropyLoss(
+            reduction="sum", use_cuda=self.config["train_config"]["use_cuda"]
```

Comment:
Do we need a separate use_cuda setting for train and eval? Any reason it can't just be a single "use_cuda" for the whole model?